### PR TITLE
build: Change x-adapter-platform build config

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-5874

Changed the configuration in the different `tsconfig` files.
Also updated `README.md` files and package descriptions

### CODE REVIEW

File: packages/x-adapter-platform/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      rules: {
+        '@typescript-eslint/ban-ts-comment': [
+          'error',
+          {
+            'ts-expect-error': false
+          }
+        ]
+      }
+    }
+  ]
```
Comment (diegopf) [inline packages/x-adapter-platform/.eslintrc.js:18]: Why do we need this?

**Review by diegopf — CHANGES_REQUESTED** [review]

File: packages/x-adapter/.gitignore [hunk 2]
```diff
@@ -1 +0,0 @@
-/types
```
Comment (LuisMartinez15) [inline packages/x-adapter/.gitignore:1]: I guess we can delete this file now.

File: packages/x-adapter-platform/tsconfig.json [hunk 3]
```diff
@@ -2,13 +2,18 @@
...
     "esModuleInterop": true,
-    "types": ["node", "jest"],
-    "lib": ["esnext"]
+    "importHelpers": true,
+    "types": ["jest"],
+    "lib": ["dom", "esnext"]
   },
   "include": ["src/**/*.ts"],
-  "exclude": ["node_modules"]
+  "exclude": ["node_modules", "**/__tests__/**"]
```
Comment (LuisMartinez15) [inline packages/x-adapter-platform/tsconfig.json:18]: Excluding the `tests` files will result in `jest` using a different `tsconfig` when running them. This is something we had to refactor in the `x-adapter` package due to miss-matching `typescript` configs between `IDE`, `tests` and `build`.

File: packages/x-adapter-platform/.eslintrc.js (hunk 1 above)
Comment (LuisMartinez15) [reply packages/x-adapter-platform/.eslintrc.js:18]: This was done in the `x-adapter` so we could use `ts-expect-error` freely while testing the schemas `types`. What I think we should question ourselves is if we are going to do some type testing here too (i.e. validating the `platform` model).

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]
Good job!

File: packages/x-adapter-platform/.eslintrc.js (hunk 1 above)
Comment (tajespasarela) [reply packages/x-adapter-platform/.eslintrc.js]: maybe we can move this to the root `.eslintrc`?  the `ts-expect-error` can be useful in more places.

I will remove here meanwhile, as it is not necessary

File: packages/x-adapter-platform/README.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
-# x-adapter-platform
+# x-adapter
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-adapter-platform/README.md]: yes, good catch, bad copy-paste

File: packages/x-adapter-platform/package.json [hunk 5]
```diff
@@ -1,22 +1,48 @@
...
+  "module": "dist/esm/index.js",
+  "types": "dist/types/index.d.ts",
+  "files": [
+    "dist"
+  ],
   "repository": {
     "type": "git",
     "url": "git+https://github.com/empathyco/x.git",
-    "directory": "packages/x-adapter-platform"
+    "directory": "packages/x-adapter"
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-adapter-platform/package.json]: yes! fixed

File: packages/x-adapter/package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@empathyco/x-adapter-next",
   "version": "0.2.0-alpha.13",
-  "description": "A search client for the Empathy search API",
+  "description": "A utils library to create a client for the any API",
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-adapter/package.json]: didn't you know the `any API`?

I'm joking, fixed.

**Review by diegopf — APPROVED** [review]

File: packages/x-adapter-platform/tsconfig.esm.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "module": "esnext",
+    "outDir": "./dist/esm",
+    "declaration": false,
+    "declarationDir": null,
+    "noEmit": false
```
Comment (LuisMartinez15) [inline packages/x-adapter-platform/tsconfig.esm.json:8]: You can remove this `noEmit` from here. We are already emitting type declarations in the `cjs` build and we are explicitly setting `declaration: false` and `declarationDir: null`.

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]

**Review by LuisMartinez15 — APPROVED** [review]
good job!

**Pull request merged**